AppCenter/AppCenter/Internals/Util/MSPerformSelectorUtil.m

```diff
@@ -27,4 +27,22 @@ + (void)performSelectorOnMainThread:(NSObject *)source withSelector:(SEL)selecto
   [invocation performSelectorOnMainThread:@selector(invoke) withObject:nil waitUntilDone:NO];
 }
 
++ (NSInvocation *)invoke:(NSObject *)source withSelector:(NSString *)selector withObjects:(NSArray *)objects {
+  SEL selectors = NSSelectorFromString(selector);
```

Comment:
Can we share some code with function above?

Reply:
I moved this method to the preprocessor. I found only such a solution.
We can't use the id or NSObject type for source, so how can we apply invoke on a primitive type. At the same time, we need to return the result of the expression if the returned type is not void. The standard objective-c solution can't do this, so I left on the preprocessor.

Reply:
Is it still true?

Reply:
No, implementation has been updated using the helper class MSPerformSelectorUtil with new implementation.